### PR TITLE
some fixes

### LINKED ISSUE #1271: [decompiler] fix "object files not named properly" error for real

### LINKED ISSUE #1307: [decompiler] run profiler and investigate slow speed
The decompiler is much slower on Windows, and the level extraction is like 10x slower than on Linux. This should be investigated a bit further because they really shouldn't be any different.

### PR DESCRIPTION
**Pull request by ManDude**
- add a `RelWithDebInfo` build for windows-clang
- change cmake flags for windows
- force windows sdk version to 7.1?
- fix some goal macros
- fix level tool being extremely slow (fixes #1307)
- fix #1271

### CODE REVIEW

File: goal_src/goal-lib.gc [hunk 1]
```diff
@@ -438,16 +438,23 @@
...
+  `(let ((it ,condition))
+      (if it
+          ,true
+          ,false
           )
       )
   )
 
+(defmacro awhen (condition &rest body)
+  "Anaphoric when, similar to Common Lisp"
```
Comment (water111) [inline goal_src/goal-lib.gc:453]: thanks dictionary, this makes it more clear
image

File: decompiler/ObjectFile/ObjectFileDB.cpp [hunk 2]
```diff
@@ -393,11 +393,14 @@ std::string ObjectFileDB::generate_obj_listing(const std::unordered_set<std::str
       result += "[\"" + pad_string(name + "\", ", 50) + "\"" +
                 pad_string(x.name_in_dgo + "\", ", 50) + std::to_string(x.obj_version) + ", " +
                 dgos + ", \"\"],\n";
-      unique_count++;
       if (all_unique_names.find(name) != all_unique_names.end() &&
           merged_objs.find(name) == merged_objs.end()) {
         lg::error("Object file {} appears multiple times with the same name.", name);
       }
+      if (merged_objs.find(name) == merged_objs.end() ||
```
Comment (water111) [inline decompiler/ObjectFile/ObjectFileDB.cpp:400]: thanks for fixing this

File: common/util/Serializer.h [hunk 3]
```diff
@@ -35,7 +36,7 @@ class Serializer {
    * later be accessed with get_save_result.
    */
   Serializer() : m_writing(true) {
-    const size_t initial_size = 32;
+    constexpr size_t initial_size = 32 * 1024 * 1024;
```
Comment (water111) [inline common/util/Serializer.h:39]: I think I have a better fix for this. There's a silly bug so it resizes on every addition :man_facepalming:

Comment (water111) [reply common/util/Serializer.h:39]: but this is fine for now

**Review by water111 — APPROVED** [review]

File: goal_src/goal-lib.gc (hunk 1 above)
Comment (ManDude) [reply goal_src/goal-lib.gc:453]: Lol, this is what I found first http://community.schemewiki.org/?anaphoric-if

Comment (ManDude) [reply goal_src/goal-lib.gc:453]: (I don't think "anaphoric when" is a lisp thing, only "anaphoric if", so I'm kind of cheating)

**Pull request merged**